src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/WorkspaceProjectStateChangeDetector.cs

```diff
@@ -461,15 +460,15 @@ private void EnqueueUpdate(Project? project, IProjectSnapshot projectSnapshot)
         _workQueue?.Enqueue(projectSnapshot.FilePath, workItem);
     }
 
-    private bool TryGetProjectSnapshot(string? projectFilePath, [NotNullWhen(true)] out IProjectSnapshot? projectSnapshot)
+    private bool TryGetProjectSnapshot(ProjectKey? projectKey, [NotNullWhen(true)] out IProjectSnapshot? projectSnapshot)
```

Comment:
It looks like every invocation of this already has to do a `ProjectKey.From`, is there a way to put it in this method instead of taking in a `ProjectKey?`?

Reply:
I explicitly want to use `ProjectKey` as the argument because in the next PR i'm changing the string that is used to construct it. Keeping this as a string parameter, means finding all calls and make sure they pass the _right_ string, which is problematic. In that same PR though, `ProjectKey.From(string)` is gone completely, so the code should be much more reliable.